app/models/insights_status.rb

```diff
@@ -0,0 +1,59 @@
+class InsightsStatus < HostStatus::Status
+  REPORT_INTERVAL  = 48.hours
+  REGISTER_DATA    = 0 # Should be registered & getting data
+  REGISTER_NO_DATA = 1 # Should be registered, but no data
+  NO_REGISTER      = 2 # Should not be registered & no data
+  NO_REGISTER_DATA = 3 # Should not be registered but getting data
+
+  def self.status_name
+    N_('Insights')
+  end
+
+  def to_label(_options = {})
+    case status
+      when REGISTER_DATA
+        N_('Recieving data')
+      when REGISTER_NO_DATA
+        N_('No data')
+      when NO_REGISTER
+        N_('Not required')
+      when NO_REGISTER_DATA
+        N_('Not installed but recieving data')
+    end
+  end
+
+  def to_global(_options = {})
+    case status
+      when REGISTER_DATA
+        ::HostStatus::Global::OK
+      when REGISTER_NO_DATA
+        ::HostStatus::Global::ERROR
+      when NO_REGISTER
+        ::HostStatus::Global::OK
+      when NO_REGISTER_DATA
+        ::HostStatus::Global::WARN
+    end
+  end
+
+  def to_status(data: nil)
+    if insights_param && data
+      REGISTER_DATA
+    elsif insights_param && !data && in_interval?
+      REGISTER_DATA
+    elsif insights_param && !data && !in_interval?
+      REGISTER_NO_DATA
+    elsif !insights_param && !data
+      NO_REGISTER
+    elsif !insights_param
+      NO_REGISTER_DATA
+    end
+  end
+
+  def in_interval?
+    (Time.now.utc - reported_at).to_i < REPORT_INTERVAL.to_i
+  end
+
+  def insights_param
+    host.host_param('host_registration_insights').nil? ? true : host.host_param('host_registration_insights')
```

Comment:
If the parameter is not set, I think the function should return `false` and not `true`.